week02/largest number.cpp

```diff
@@ -0,0 +1,29 @@
+// Time:  O(nlogn)
+// Space: O(1)
+
+class Solution {
+public:
+    string largestNumber(vector<int>& nums) {
+
+        //sorted according to each character's Unicode code point value
+        //a=The first element for comparison.
+        //b=The second element for comparison.
+
+        sort(nums.begin(),nums.end(),[&](const int& a,const int& b){
+
+            string s1 = to_string(a) + to_string(b);
+            string s2 = to_string(b) + to_string(a);
+            return s1 > s2;
+        });

```

Comment:
This is a C++ lambda expression. Are you sure you understand how lambda works? I would recommend not taking a solution from any discussion forum and using it without understanding properly.

Reply:
To be honest, I did not know much about the lambda function before this solution. I first learnt about it for this particular solution, then I use it here. But I do not know much more about this function. I understood the problem, but, I could not figure out how to solve it in a optimal way, that's why, Yes, I took the concept of this solution from google and leetcode discussion. Then I learnt what I did not know, then I used it.